### PR TITLE
WIP: Find correspondences across several datasets

### PR DESCRIPTION
**Pull request by XachaB**
This is a work in progress, do not merge. 
Things to do:

- [ ] The code currently relies on a hard coded list of datasets, and fakes the lexicore interface. This is a temporary way to not be dependent on the timeline for lexicore. Instead, replace by actual lexicore.
- [ ] Implement alternatives to genera to find sub-groups in each family
- [ ] Use a better cognate detection method
- [ ] Maybe implement something to have a coarser sound classification than BIPA

### CODE REVIEW

**Comment by LinguList** [comment]
How do I call the code, can you give an example?

**Comment by XachaB** [comment]
Sure, it is pretty similar to what you originally wrote:

~~~
cldfbench lexitools.correspondences --model asjp --cutoff 2 --threshold 1 --concepts Holman-2008-40
~~~

-  `--model` can be "asjp" or "bipa" (this is used to downscale phonemes)
- `--concepts` limits to a list of concepts. Here using Holman-2008-40
- `--cutoff` lets you pick how many times a correspondence must appear to be kept
- `--threshold` is the similarity threshold between pairs of words

This will run on several datasets. To run on a single dataset:

~~~
cldfbench lexitools.correspondences --model asjp --cutoff 2 --threshold 1 --concepts Holman-2008-40 --dataset asjp
~~~

I think "asjp" is the only case where we may want to run it on a single dataset.

**Comment by LinguList** [comment]
Okay, now the question: "several datasets": what are these datasets, all 
those that are installed? or how do you determine them?

**Comment by XachaB** [comment]
As I said above, this is what simulates "lexicore" (the` --dataset` argument defaults to "lexicore"). I implemented this before lexicore was ready, and needed something which would more or less run on the same set of datasets, and be easily swappable with the real lexicore interface once lexicore is ready. This is not code that is intended to stay, it is merely a work around to make sure the sound correspondence work can go forward as much as possible without waiting for the technical details of lexicore. As soon as lexicore is ready I can very easily swap it in.

The list I have currently is the below, which I got from the google sheet. Some datasets do not install/read properly.

~~~
["aaleykusunda", "abrahammonpa", "allenbai",
                        "backstromnorthernpakistan", "bdpa", "beidasinitic",
                        "birchallchapacuran", "bodtkhobwa", "bowernpny", "cals",
                        "castrosui", "castroyi", "chaconarawakan",
                        "chaconbaniwa", "chaconcolumbian", "chenhmongmien",
                        "chindialectsurvey", "clarkkimmun", "dravlex",
                        "dunnaslian", "galuciotupi", "halenepal",
                        "hantganbangime", "housinitic", "hsiuhmongmien",
                        "hubercolumbian", "ivanisuansu", "joophonosemantic",
                        "lieberherrkhobwa", "listcognatebenchmark",
                        "liusinitic", "marrisonnaga", "mcelhanonhuon",
                        "mitterhoferbena", "mortensentangkhulic",
                        "naganorgyalrongic", "northeuralex", "robinsonap",
                        "sagartst", "savelyevturkic", "sohartmannchin", "suntb",
                        "tppsr", "transnewguineaorg", "tryonsolomon",
                        "vanbikkukichin", "walworthpolynesian", "wangbai",
                        "wichmannmixezoquean", "wold", "yanglalo",
                        "zgraggenmadang", "gerarditupi", "polyglottaafricana",
                        "peirosaustroasiatic", "deepadungpalaung", "zhaobai"]
~~~

File: src/lexitools/commands/correspondences.py [hunk 1]
```diff
@@ -2,85 +2,150 @@
...
+                print("Failed to load", name)
+
+    def iter_table(self, table_name):
+        for name, ds in self.datasets.items():
+            for row in ds[table_name]:
+                row["dataset"] = name
+                yield row
+
+    def get_table(self, table_name):
+        return list(self.iter_table(table_name))
```
Comment (XachaB) [inline src/lexitools/commands/correspondences.py:53]: This is what simulates the lexicore interface. Given a list of datasets, it simply installs each dataset (if not already installed)  and then chains the dataset tables together. This should be very similar to the interface @tresoldi wrote for the python code to load the sql database from lexicore.

**Comment by LinguList** [comment]
I think, @XachaB, you can also just merge this for now, I'd then later check quietly and provide updates and other suggestions. The current solution you propose for cloning is okay, I'd say.

File: src/lexitools/commands/correspondences.py (hunk 1 above)
Comment (tresoldi) [reply src/lexitools/commands/correspondences.py:53]: Yes, it is exactly like the interface in `lexicore.py`.

**Pull request merged**